docs/integrations/monitoring/datadog.md

```diff
@@ -2,31 +2,115 @@
 
 You can add operational visibility to your [Rack](../../reference/primitives/rack) with Datadog.
 
+## Configure Kubectl to Point at Your Rack
+
+Convox allows you to securely connect your Kubectl to your Convox created Kubernetes cluster by exporting a [Kubeconfig](../../reference/cli/rack#rack-kubeconfig) that will connect you to a Kubernetes API Proxy running inside your Rack. This allows you to use Kubectl without directly exposing the credentials for your Kubernetes cluster. For example if your Rack is called `myrack` you could point your local Kubectl to your Rack cluster as follows
+
+```
+$ convox rack kubeconfig /tmp/myrack-config
+$ export KUBECONTROL=/tmp/myrack-config
+```
+
+This will export the proxy configuration to a temporary file and then point your local Kubecontrol at that location so you can connect to your Rack's cluster. You will need to perform this step before you can execute any Kubectl commands against your cluster.
+
 ## Deploy the Datadog Agent
 
-You can deploy the datadog agent as a Convox [App](../../reference/primitives/app) with
-a simple `convox.yml`:
-
-    services:
-      datadog:
-        agent: true
-        environment:
-          - DD_API_KEY
-          - DD_APM_ENABLED=true
-        image: datadog/agent:latest
-        ports:
-          - 8125/udp
-          - 8126/tcp
-        privileged: true
-        scale:
-          cpu: 128
-          memory: 128
-        volumes:
-          - /proc/:/host/proc/
-          - /sys/fs/cgroup/:/host/sys/fs/cgroup/
-          - /var/run/docker.sock:/var/run/docker.sock
-
-> Make sure to set the `DD_API_KEY` environment variable to your Datadog API key.
+Once you have Kubectl pointing at your Rack you can deploy the datadog agent as a Kubernetes Daemonset. The following is based on the [DataDog Documentation](https://docs.datadoghq.com/agent/kubernetes/?tab=daemonset) so please refer back there for any specific tweaks you may want to make.
+
+## Configure Agent Permissions
```

Comment:
I'm wondering if at this point we should just link to the Datadog instructions so we don't have to stay in sync if they change things.

Reply:
I wanted to do that originally as well but I found the datadog documentation to be a little bit confusing and missing a few important pieces (like getting the base64 encoded key). I like to use myself as reasonable proxy for your average Convox user and I find this step by step to be an improvement.

Reply:
Does Step 2 in their guide not create the base64-encoded secret?

Reply:
The Datadog instructions seem to change quite frequently and I worry about these instructions getting out of date. Especially since you are linking to URLs for manifests where the instructions may change.

Reply:
Just read over their docs more and I can see what you're talking about. They recreate/clobber the secret in their manifest files. That's a shame. Your call. I think we're going to pretty quickly get out of date having our own fork of the instructions but seems like theirs don't work out of the box.

Reply:
Yeah I struggle with the exact same concerns. I would have loved to just link to their docs but they hung me up and I don't want people to silently struggle. That's why I comprised by including the links to their docs in a few places in case things get out of sync. I vote we merge as is to meet the immediate needs of people who have requested updated instructions.

Reply:
Yeah, I'd say we should merge as well. If Datadog improve, we can adapt. If they just change, but stay broken, we'll have to adapt anyway. 🤷‍♂️